### PR TITLE
Bold nested search

### PR DESCRIPTION
**Pull request by Wilker0725**
**Changes**
Description of changes...
* Bold text if matches the search term
* Selectable on title label itself
* Update stories and types description

- [delete] branch

- [WARNING] Description of change in `NestedSelect`

### CODE REVIEW

File: stories/form/form-nested-select/form-nested-select.stories.mdx
Comment (qroll) [inline stories/form/form-nested-select/form-nested-select.stories.mdx:1]: to import the required components
`import { Canvas, Meta, Story } from "@storybook/addon-docs";`

File: stories/form/form-nested-select/props-table.tsx [hunk 1]
```diff
@@ -11,7 +11,116 @@ const DATA: ApiTableSectionProps[] = [
                 name: "options",
                 mandatory: true,
                 description: "A list of options that a user can choose from",
-                propTypes: ["T[]"],
+                propTypes: ["L1OptionProps<V1, V2, V3>[]"],
+            },
+            {
+                name: "selectedKeyPath",
+                description:
+                    "Specifies key path to select particular option label",
```
Comment (qroll) [inline stories/form/form-nested-select/props-table.tsx:19]:
```suggestion
                    "The key path of the selected option",
```

File: stories/form/form-nested-select/props-table.tsx [hunk 2]
```diff
@@ -11,7 +11,116 @@ const DATA: ApiTableSectionProps[] = [
...
+            {
+                name: "readOnly",
+                description:
+                    "Indicates if the component has a read only state and selection is not allowed",
+                propTypes: ["boolean"],
+            },
+            {
+                name: "mode",
+                description:
+                    "The visual state to represent the dropdown behavior when dropdown is opened",
```
Comment (qroll) [inline stories/form/form-nested-select/props-table.tsx:63]:
```suggestion
                    "Determines if items are expanded or collapsed when the dropdown is opened",
```

File: stories/form/form-nested-select/props-table.tsx
Comment (qroll) [inline stories/form/form-nested-select/props-table.tsx:1]: to include `selectableTitle`, `valueToStringFunction`, and `hideNoResultsDisplay` props

File: src/shared/nested-dropdown-list/nested-dropdown-list.tsx
Comment (qroll) [inline src/shared/nested-dropdown-list/nested-dropdown-list.tsx:1]: side question, is there no max height needed on desktop/tablet?

**Review by qroll — CHANGES_REQUESTED** [review]

File: src/shared/nested-dropdown-list/list-item.tsx [hunk 3]
```diff
@@ -73,25 +94,28 @@ export const ListItem = <V1, V2, V3>({
...
+        if (!item.isSearchTerm) {
+            return <>{item.label}</>;
         }
 
-        return StringHelper.shouldTruncateToTwoLines(
-            displayText,
-            widthOfElement
+        const label = item.label;
+        const searchTerm = searchValue.toLowerCase().trim();
+        const startIndex = label.toLowerCase().indexOf(searchTerm);
```
[the remark this answers is not in the document]
Comment (Wilker0725) [reply src/shared/nested-dropdown-list/list-item.tsx:110]: wa! good catch

**Review by qroll — APPROVED** [review]

**Pull request merged**